### PR TITLE
Remote ID Disabling

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: The Captain and Head of Personnel can now use an ID Console to remotely disable IDs. On top of ID access, doing so requires an ID of a higher tier than the target id (gold > silver, silver > basic). Agent IDs are unlisted and immune.
/:cl:

# How it works
The system requires an ID that:
- Has ID Modification access
- Is of higher tier than the target ID

Centcomm > Gold > Silver > Normal

Agent IDs are unlisted and immune, giving them an additional advantage: safe access stealing.

To prevent trolling by grabbing the captain's spare and disabling everyone's id, there's a cooldown between disabling (not re-enabling) IDs. It's currently set at 15s, but it's easily adjusted.

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 

## What does this add to the game
- It makes demotion feasible. Currently you have to catch the person to demote, which is likely running away to not lose their access, strip them of their id, drag both to the ID console and remove the access.
This is likely to result in escalation on both parts for an action that should be a basic deterrent tool for heads of staff.
With remote disabling, you can cut their access until they come back to the HoPline to get it restored (after it has been edited).

- It gives a countermeasure to all-access infection. When an assistant grabs hold of an all-access id, he can freely make copies and distribute them, usually resulting in complete disorder that the heads can do little about. With remote disabling, you can disable the copies (since the captain's spare would be immune). This is not foolproof, since someone can simply rename their ID, lay low, and dodge the disabling sweep.

- It makes sense to have as a system for high-value cards. IRL you can call your bank to disable your credit card if it gets stolen. If you somehow get cloned after being murdered and robbed, you can ask the HoP to disable your old card. This would be countered by hiding bodies, or at least disfiguring them as to not give hints about whose id got stolen.

### CODE REVIEW

**Comment by XDTM** [comment]
Closed for freeze

**Comment by Bawhoppen** [comment]
bad idea for reasons I'm sure someone like robustin will articulate impeccably

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -84,6 +84,21 @@
 	var/registered_name = null // The name registered_name on the card
 	var/assignment = null
 	var/dorm = 0		// determines if this ID has claimed a dorm already
+	var/disabled = FALSE // if it has been remotely disabled
+	var/tier = 1 // level of the card for disabling purposes
```
Comment (coiax) [inline code/game/objects/items/weapons/cards_ids.dm]: These tiers should be DEFINES.

Comment (XDTM) [inline code/game/objects/items/weapons/cards_ids.dm]: Tiers are checked by comparing the values, and 4 < 1 is more clear than TIER_BASIC < TIER_CENTCOMM

**Comment by BeeSting12** [comment]
I don't like this idea much, coming from someone who deals with this shit fairly often.

**Comment by PKPenguin321** [comment]
Hesitant

**Comment by octareenroon91** [comment]
This is extreme power for a traitor hop.

Captain's spare should possibly be downgraded to silver or electrum

**Comment by ExcessiveUseOfCobblestone** [comment]
If the person never returns to the department isn't that just as good as an official demote? If they DO come back just subdue them with your infinite,no charge baton and find your nearest cablecuffs / straight jackets.

**Comment by Repukan** [comment]
How am I supposed to beat them half to death if I catch them now?

**Comment by uraniummeltdown** [comment]
Wayyy too much power

**Comment by Dorsisdwarf** [comment]
What a moronic idea

**Comment by kevinz000** [comment]
hugbox

**Comment by XDTM** [comment]
welp

**Comment by coiax** [comment]
I thought it was cool, but apparently everyone likes assistant all access rounds

**Comment by octareenroon91** [comment]
I also thought it was cool but needed more balance planning or testing before committing it to part of the game.

**Pull request closed without merging**